main.go

```diff
@@ -107,28 +108,41 @@ func ProcessExportsSince(wh warehouse.Warehouse, tableColumns []string, since ti
 func ProcessFilesIndividually(wh warehouse.Warehouse, tableColumns []string, fs *fullstory.Client, exports []fullstory.ExportMeta) (int, error) {
 	for _, e := range exports {
 		log.Printf("Processing bundle %d (start: %s, end: %s)", e.ID, e.Start.UTC(), e.Stop.UTC())
-		filename := filepath.Join(conf.TmpDir, fmt.Sprintf("%d.csv", e.ID))
 		mark := time.Now()
-		outfile, err := os.Create(filename)
-		if err != nil {
-			log.Printf("Failed to create tmp file: %s", err)
-			return 0, err
-		}
-		defer os.Remove(filename)
-		defer outfile.Close()
-		csvOut := csv.NewWriter(outfile)
+		var filename string
+		var statusMessage string
+		if conf.Local.SaveAsJson {
+			filename = filepath.Join(conf.TmpDir, fmt.Sprintf("%d.json", e.ID))
+			defer os.Remove(filename)
+			writtenCount, err := WriteBundleToJson(fs, e.ID, filename)
```

Comment:
It was mainly done that way because WriteBundleToCSV already existed, and WriteBundleToJson didn't, and it seemed like it would be more cumbersome to try and combine the two (they do both share the call to getExportData, and gzip reader, but that's about it). Currently the decision between the two only gets made in ProcessFilesIndividually (since it doesn't really make sense to group json files by day), but in the future it could also be made elsewhere (if needed).

Reply:
sgtm also as Jess mentioned seems like there will be future work on Hauser, which hopefully will take care of this!